### DISCUSSION TITLE
Support for Typst

### DISCUSSION THREAD

**Discussion by nekomiya-kasane** [outline/outline#7629] [Ideas]
[Typst](https://github.com/typst/typst) is a new markup-based typesetting system, much easier than latex grammar, I wonder is it possible to integrate it into outline? Supporting math equations in typst lang would also be great. Personally I'm tired of typing backslashes... Additionally, if typst can be supported in the future, drawing vector graphs with codes in paragraphs would also be possible, just as we do drawing with typst.

**Comment by xhyrom** [comment]
This would be great, looking forward for this feature :)

**Comment by DorianBourgeoisat** [comment]
I volunteer to implement a typst export feature, if I have the greenlight.
Typst is a new typesetting language, with the simplicity of markdown and the expressivity of latex (or at least close to).
Outline having a typst export would allow for companies and non-profits using Outline for collaborative note-taking for minutes, and exporting it to typst for templating and corporate identity.

As an exmple of typst syntax (the markup part because that's what we'd be exporting, with examples of a show rule to show the expressivity) :

```typst
#show heading: set text(fill: red) // sets headings fill color to red
#show strong: it => { // this show rule prefixes strongly emphasised text with "VERY important : ", VERY, being itself emphasised
    [_VERY_ important : ] + it
}

= Heading 1
_Emphasised text_
== Heading 2 <head2>
*Strongly emphasised text*

 - List of things 
 - Go to @head2

```

**Reply by tommoor** [reply]
I'm reticent because this feature has never been requested up until now – I do not believe that core users of Outline are also Typst users. If the export feature has a small maintenance burden by using an existing library for example it _may_ be acceptable.

What approach are you thinking of?
